dbms/src/Interpreters/Settings.h

```diff
@@ -306,6 +306,7 @@ struct Settings
     M(SettingBool, cancel_http_readonly_queries_on_client_close, false, "Cancel HTTP readonly queries when a client closes the connection without waiting for response.") \
     M(SettingBool, external_table_functions_use_nulls, true, "If it is set to true, external table functions will implicitly use Nullable type if needed. Otherwise NULLs will be substituted with default values. Currently supported only for 'mysql' table function.") \
     M(SettingBool, allow_experimental_data_skipping_indices, false, "If it is set to true, data skipping indices can be used in CREATE TABLE/ALTER TABLE queries.")\
+    M(SettingBool, allow_short_circuit_logic_expressions, false, "If it is set to true, logic expressions will be optimized by short circuit, but this may make the alias lost by optimization")\
```

Comment:
Lost alias shouldn't be an option. We can push aliases to `WITH` section of the query.

Reply:
`select a  from table where 1 = 1 or  (( device like '%android%' ) as t)`
will be optimized to 
`select a  from table where 1`,  the device column will never be read,  should we keep the alias `t` ?

Reply:
If you write
```
with ( device like '%android%' ) as t select a from table where 1
```
column `device` will not read.